lib/components/term.js

```diff
@@ -200,6 +200,12 @@ export default class Term extends Component {
     if (this.props.customCSS !== nextProps.customCSS) {
       this.term.prefs_.set('user-css', this.getStylesheet(nextProps.customCSS));
     }
+
+    if (this.props.bell === 'SOUND') {
+      this.term.prefs_.set('audible-bell-sound', this.props.bellSoundURL);
+    } else {
+      this.term.prefs_.set('audible-bell-sound', '');
+    }
```

Comment:
Where we initialize the term and its preferences. `componentWillReceiveProps` only gets invoked when `props` change, not on initial render. So the sound pref wouldn't be honored unless you change the config.